include/GIL/Instructions/StructCreateInst.hpp

```diff
@@ -0,0 +1,136 @@
+    friend TrailingArgs;
+
+    unsigned _structCreateInstCount;
+
+    // Method required by llvm::TrailingObjects to determine the number
+    // of trailing objects.
+    size_t numTrailingObjects(typename TrailingArgs::OverloadToken<Value>) const
+    {
+        return _structCreateInstCount;
+    }
+
+    /// @brief Constructs a StructCreateInst object.
+    ///
+    /// @param structType The type of the structure being created.
+    /// @param members An array of values for each member of the structure.
+    StructCreateInst(Type structType, llvm::ArrayRef<Value> members)
+        : AggregateInst(InstKind::StructCreateInstKind)
+        , _structType(structType)
+        , _structCreateInstCount(members.size())
+    {
+        assert(llvm::isa<types::StructTy>(structType.getType()));
+        std::uninitialized_copy(
+            members.begin(), members.end(),
+            this->template getTrailingObjects<Value>()
+        );
+    }
+
+public:
+    /// @brief Constructs a StructCreateInst object.
+    ///
+    /// @param structType The type of the structure being created.
+    /// @param members An array of values for each member of the structure.
+    static StructCreateInst *create(
+        llvm::BumpPtrAllocator &alloc, Type structType,
+        llvm::ArrayRef<Value> members
+    )
+    {
+        auto totalSize = totalSizeToAlloc<Value>(members.size());
+        void *mem = alloc.Allocate(totalSize, alignof(StructCreateInst));
+
+        return new (mem) StructCreateInst(structType, members);
+    }
+
+    /// @brief Sets the structure type.
+    ///
+    /// @param value The new structure type.
+    void setStruct(Type value) { this->_structType = value; }
+
+    /// @brief Gets the structure type.
+    ///
+    /// @return The structure type.
+    Type getStruct() const { return _structType; }
+
+    /// @brief Gets the values of the structure members.
+    ///
+    /// @return An array containing the values of all structure members.
+    llvm::ArrayRef<Value> getMembers() const
+    {
+        return { getTrailingObjects<Value>(), _structCreateInstCount };
+    }
+
+    /// @brief Gets the number of operands required by this instruction.
+    ///
+    /// @return 1 (for the structure type) plus the number of member values.
+    size_t getOperandCount() const override
+    {
+        return 1 + _structCreateInstCount;
+    }
+
+    /// @brief Gets the operand at the specified index.
+    ///
+    /// @param index The index of the operand (0 for structure type, 1+ for
+    /// member values).
+    /// @return The operand at the specified index.
+    Operand getOperand(size_t index) const override
+    {
+        assert(index < getOperandCount() && "Operand index out of range");
+        if (index == 0)
+            return Operand(_structType);
+        llvm::ArrayRef<Value> members
+            = { getTrailingObjects<Value>(), _structCreateInstCount };
+        return Operand(members[index - 1]);
```

Comment:
```suggestion
        return getMembers()[index - 1];
```
Operand is an implicit conversion